devito/finite_differences/coefficients.py

```diff
@@ -220,24 +231,26 @@ def generate_subs(i):
 def default_rules(obj, functions):
 
     def generate_subs(deriv_order, function, dim):
+        extracted_dim = retrieve_dim(dim)
```

Comment:
The naming here could do with a review, `extracted_dim = retrieve_dim(dim)` is a bit weird.
`dim = retrieve_dimension(some_suitable_name)` would be better.